### DISCUSSION TITLE
(definitive?) DomRect fingerprinting protection

### DISCUSSION THREAD

**Discussion by uazo** [bromite/bromite#2128] [Ideas]
@csagan5 before going on holiday, I ask you for your opinion about a possible patch that I would like to finish.

the idea is to take advantage of the possibility of changing the resolution in dpi of the screen in blink with the aim of falsifying all domrect value read by javascript (which in reality will be not false, but different each navigation).
plus, if I can also change the base percentage of the zoom, for the user will be the same as before.

I'll explain. assuming that the system screen has a dip scale of 3.5 on a surface area of ​​1440 pixels, blink shows the user an "effective" resolution of 412 pixels. the idea is that, by playing with scaling, I can bypass all possible coordinate-based fingerprinting a priori, whatever they are, present and future, I think, as well as those read with webgl or css or similar systems.

for now I have managed to modify the scale with a random factor for each webcontents (i.e. at the level of tab, not by site) because switching from one site to another by modifying the scaling currently produces me an annoying glitch that I haven't been able to eliminate yet.

moreover, I have not yet been able to understand (at the moment :) how to manage the zoom, because by changing the dpi I would also like to change the zoom ratio too, with a factor that restores the "visible" dpi to the original value, so that the final effect is to be transparent to the user.
and finally of course hiding the information to javascript that would allow it to be understood.

so, what do you think? Is it something that you think would be worth trying to do or do you think there is something that escapes me for now?

**Comment by csagan5** [comment]
It would be a clever approach; if it introduces enough noise to make fingerprints useless (and it is not reversible) we could introduce it.

However I would not call it "definitive", improvements are always possible and new fingerprinting techniques as well.

**Comment by csagan5** [comment]
Related: https://github.com/z0ccc/Vytal
